docs/tutorial/fdp/index.rst

```diff
@@ -0,0 +1,179 @@
+complete command can be run like:
+
+.. literalinclude:: 400_xnvme_log_fdp_events.cmd
+   :language: bash
+
+The command should produce output similar to:
+
+.. literalinclude:: 400_xnvme_log_fdp_events.out
+   :language: bash
+
+Set and get-feature
+-------------------
+
+There are 2 new set and get-feature commands, introduced with FDP. These are
+Flexible data placement which controls operation of FDP capability in the
+specified endurance group, FDP events which controls if a controller generates
+FDP events associated with a specific reclaim unit handle.
+
+xNVMe does not support namespace management commands. Thus we cannot enable or
+disable FDP by sending set-feature command to the endurance group, as it
+requires deletion of all namespaces in that endurance group. However you can
+check the FDP capability by sending a get-feature command. The command can be
+run like:
+
+.. literalinclude:: 010_xnvme_feature_get.cmd
+   :language: bash
+
+The command should produce output similar to:
+
+.. literalinclude:: 010_xnvme_feature_get.out
+   :language: bash
+
+Command dword 12 controls whether you want to enable or disable FDP events.
+You will have to specify number of events to enable or disable and placement
+handle associated with it. These will be part of feat field. This will require
+you to specify the size in bytes of data buffer. To enable all the fdp events
+you can run command like:
+
+.. literalinclude:: 020_xnvme_set_fdp_events.cmd
+   :language: bash
+
+The command should produce output similar to:
+
+.. literalinclude:: 020_xnvme_set_fdp_events.out
+   :language: bash
+
+You can get the status of supported fdp events. The command dowrd 11 remains
+same as the set-feature command. You can run the command like:
+
+.. literalinclude:: 030_xnvme_feature_get.cmd
+   :language: bash
+
+The command should produce output similar to:
+
+.. literalinclude:: 030_xnvme_feature_get.out
+   :language: bash
+
+I/O management
+--------------
+
+Two I/O management commands are introduced with FDP. These are I/O management
+send and I/O management receive.
+
+I/O management receive supports reclaim unit handle status command. You will
+have to specify the number of recalim unit handle status descriptors to fetch.
+You can run the command like:
+
+.. literalinclude:: 040_xnvme_fdp_ruhs.cmd
+   :language: bash
+
+The command should produce output similar to:
+
+.. literalinclude:: 040_xnvme_fdp_ruhs.out
+   :language: bash
+
+I/O management send supports reclaim unit handle update command. You will have
+to specify placement identifier for this. You can run the command like:
+
+.. literalinclude:: 050_xnvme_fdp_ruhu.cmd
+   :language: bash
+
+The command should produce output similar to:
+
+.. literalinclude:: 050_xnvme_fdp_ruhu.out
+   :language: bash
+
+xnvme ioengine
+--------------
+
+FIO's xnvme ioengine provides fdp support since 3.35 release.
+For more information on the fdp specific options you can look at:
```

Comment:
```suggestion
FIO's xNVMe ioengine provides FDP support since the 3.35 release.
For more information on the FDP specific options you can look at:
```